### ISSUE TITLE
drop into repl after successful loading

### ISSUE THREAD

**Issue by adrian-gierakowski** [ndmitchell/ghcid#250]
is it possible to drop into interactive repl session after successful loading/reloading?

thanks!

**Comment by ndmitchell** [comment]
Nope, and it would be hard to do since ghcid assumes it has complete control over the session, so has done crazy things like giving it a weird prompt etc. It's not completely infeasible, since ghcid could give you its own session and proxy them on, but its not in the plans.

**Comment by adrian-gierakowski** [comment]
Thank you for explanation.

When you say "its not in the plans" is it because you don't think it would be useful, or because you'd rather focus hour efforts somewhere else? If the latter, would you accept a PR implementing it?

**Comment by ndmitchell** [comment]
My focus is elsewhere, I think it could be quite a significant piece of work, I think there are a lot of trade offs, and I'm not convinced the end result will be that nice. That said, if you make a PR, the PR doesn't overly complicate stuff, and it works reasonably enough, I'm happy - it is something quite a lot of people ask for.

That said, I do wonder if something that reloads would be better integrated into ghci itself, since you are taking the prompt bits from ghci and the reloading from ghcid - and the combination involves much more ghci.

**Comment by freckletonj** [comment]
Glad I found this issue, I totally thought `ghcid` was a repl enhancement, and was pulling my hair out trying to solve why it was broken.

> "GHCi as a daemon" or "GHC + a bit of an IDE".
> it opens ghci and runs :reload

So `ghcid` basically just compiles your code to give you ghc's error output, only, it does so automatically, and actually interpreting is faster to do than compiling, so it does that instead.

**Comment by freckletonj** [comment]
One more thing I just discovered surfing the issues, is the flag `--allow-eval`.

**Comment by ndmitchell** [comment]
Thanks for letting me know @freckletonj - it's not an unreasonable confusion - I've updated the README to say you can't use the ghci it spawns: https://github.com/ndmitchell/ghcid/blob/master/README.md. There is an active ticket for documenting `--allow-eval`.

**Comment by docteurklein** [comment]
my 2 cents on this topic: you can use `intofiywait` in conjunction with `tmux` to automate pretty much anything and emulate a part of what ghcid does:

```
inotifywait -e create -e modify -r -m --format='%w%f' src | xargs -L1 -I{} tmux send-keys -t '1.2' 'C-l' ':r' enter
```
would clear the repl (`C-l`) and reload the sources (`':r' enter`) everytime something changed in the `src` folder.

You first need to run your repl in a tmux pan (in my example located at `1.2`).

Hope it helps :)

**Comment by adrian-gierakowski** [comment]
@docteurklein thanks for the tip!
